omp/factorization/cholesky_kernels.cpp

```diff
@@ -248,34 +250,42 @@ void factorize(std::shared_ptr<const DefaultExecutor> exec,
     const auto row_ptrs = factors->get_const_row_ptrs();
     const auto cols = factors->get_const_col_idxs();
     const auto vals = factors->get_values();
+    vector<int> ready(num_rows, {exec});
+#pragma omp parallel for schedule(monotonic : auto)
     for (size_type row = 0; row < num_rows; row++) {
```

Comment:
Just a question: could you start with the rows that have no dependencies here? I am not sure, are those the roots in the elimination forest? I see that for GPUs that doesn't really make a difference because you start many many threads, but maybe here it would?